streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java

```diff
@@ -67,6 +68,68 @@
     }
 
     @Override
+    public KeyValueIterator<Bytes, byte[]> all() {
+        final List<Segment> searchSpace = segments.allSegments();
+        
+        final Segment minSegment = segments.minSegment();
+        final Segment maxSegment = segments.maxSegment();
+        final Bytes minKey = minSegment.first().key;
+        final Bytes maxKey = maxSegment.last().key;
+        
+        final long minTimestamp = keySchema.segmentTimestamp(minKey);
+        final long maxTimestamp = keySchema.segmentTimestamp(maxKey);
+        final Bytes keyFrom = WindowStoreUtils.bytesKeyFromBinaryKey(minKey.get());
+        final Bytes keyTo = WindowStoreUtils.bytesKeyFromBinaryKey(maxKey.get());
+        
+        return new SegmentIterator(searchSpace.iterator(),
+                                   keySchema.hasNextCondition(keyFrom, keyTo, minTimestamp, maxTimestamp),
+                                   minKey, maxKey);
+    }
+    
+    @Override
+    public KeyValueIterator<Bytes, byte[]> fetchAll(final long timeFrom, final long timeTo) {
+        final Segment minSegment = segments.getMinSegmentGreaterThanEqualToTimestamp(timeFrom);
+        final Segment maxSegment = segments.getMaxSegmentLessThanEqualToTimestamp(timeTo);
+        
+        long minTimestamp = Long.MAX_VALUE;
+        long maxTimestamp = 0;
+        Bytes keyFrom = new Bytes(new byte[0]);
+        Bytes keyTo = null;
+        
+        final KeyValueIterator<Bytes, byte[]> iteratorMin = minSegment.all();
+        while (iteratorMin.hasNext()) {
```

Comment:
Do we need this loop? Is there any reason why `keyFrom` can't just stay as `new Bytes(new byte[0])`?